\section{Introduction}

The motif finding problem involves identifying recurring patterns in biological sequences that often represent functional elements such as transcription factor binding sites. It is a fundamental problem in bioinformatics, with applications in gene regulation, protein-DNA interactions, and other biological processes. The discovery of motifs is crucial for understanding the regulatory mechanisms of genes
and proteins, and has implications in drug discovery and disease treatment. It is an NP-hard problem, which is made difficult by a number of biological factors.

Motif finding involves searching for many small sequences among long DNA sequences. Moreover, motifs are degenerate, as there are variations in their nucleotide sequences caused during cell division and mutation. Data for DNA sequences is often noisy and contains errors, making it difficult to identify them. Additionally, motifs can be located in different positions in different sequences, and the length of the motif is often unknown. Each sequence may contain multiple motifs; moreover they may be overlapped or nested. Thus, heuristic algorithms are used to solve the motif finding problem.

In this report we present a comparative analysis of two prominent motif finding algorithms: Randomized Motif Search (RMS) and Gibbs Sampler. Additionally, we evaluate the effectiveness of two widely used motif discovery tools, MEME and STREME, which implement these algorithms.
